### PR TITLE
[REFACTORING] [CFDS] [TRAH] shontzu/CFDS-3634/add-compare-accounts-page

### PR DESCRIPTION
**Pull request by shontzu-deriv**
### [REFACTORING] Add compare accounts popup as we have in production

a [new repository](https://github.com/shontzu-deriv/traders-hub) has been initiated for tradershub v2 

as the next phase of FE refactoring, tradershub v2 will be deprecated from legacy codebase 

we need to migrate all tradershub-v2-related code from [legacy codebase](http://hub.com/shontzu-deriv/deriv-app) to tradershub's [new standalone repository](https://github.com/shontzu-deriv/traders-hub)

the scope of this card is to migrate the compare-accounts related code

image

### CODE REVIEW

File: src/cfd/components/CompareAccountsCarousel/CompareAccountsCarouselButton.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    onClick: () => void;
+};
+
+const CFDCompareAccountsCarouselButton = ({ enabled, isNext = false, onClick }: TPrevNextButtonProps) => (
+    <button
+        className={twMerge(
+            'bg-system-light-primary-background z-10 absolute flex items-center justify-center top-1/2 cursor-pointer w-40 h-40 rounded-[50%] disabled:opacity-8 disabled:hidden border-0 shadow-7',
+            isNext && 'right-16',
+            !isNext && 'left-16'
+        )}
```
Comment (hasan-deriv) [inline src/cfd/components/CompareAccountsCarousel/CompareAccountsCarouselButton.tsx:17]:
```suggestion
        className={twJoin(
            'bg-system-light-primary-background z-10 absolute flex items-center justify-center top-1/2 cursor-pointer w-40 h-40 rounded-[50%] disabled:opacity-8 disabled:hidden border-0 shadow-7',
            isNext && 'right-16',
            !isNext && 'left-16'
        )}
```

File: src/cfd/screens/CFDCompareAccounts/CompareAccountsPlatformLabel.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+const CompareAccountsPlatformLabel = ({ platform }: TCompareAccountsPlatformLabel) => {
+    const platformType = getPlatformType(platform);
+
+    return (
+        <div
+            className={twMerge(
+                CompareAccountsPlatformLabelClass({
+                    background: platformType,
+                } as unknown as TCompareAccountsPlatformLabelClassProps)
+            )}
```
Comment (hasan-deriv) [inline src/cfd/screens/CFDCompareAccounts/CompareAccountsPlatformLabel.tsx:29]: I think `twJoin` would work here.  An why this `as unknown as TCompareAccountsPlatformLabelClassProps`

File: src/cfd/screens/CFDCompareAccounts/InstrumentsLabelHighlighted.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+const InstrumentsLabelHighlighted = ({ marketType, platform, shortCode }: TInstrumentsLabelHighlighted) => {
+    const { data: activeDerivTrading } = useActiveDerivTradingAccount();
+    const { regulationFlags } = useRegulationFlags();
+    const { isEU: isEuRegion } = regulationFlags;
+    const isDemo = activeDerivTrading?.is_virtual;
+    const iconData = [...getHighlightedIconLabel(platform, isEuRegion, marketType, shortCode)];
+
+    return (
+        <div
+            className={twMerge('flex flex-col pt-20 px-14 lg:px-18 gap-4', isDemo && 'pt-16')}
```
Comment (hasan-deriv) [inline src/cfd/screens/CFDCompareAccounts/InstrumentsLabelHighlighted.tsx:24]:
```suggestion
            className={twJoin('flex flex-col px-14 lg:px-18 gap-4', isDemo ? 'pt-16' : 'pt-20')}
```

Can avoid `twMerge` here

File: src/cfd/screens/CFDCompareAccounts/CompareAccountsCard.tsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    platform: TPlatforms.All;
+    shortCode: THooks.AvailableMT5Accounts['shortcode'];
+};
+
+const CompareAccountsCard = ({ isAccountAdded, marketType, platform, shortCode }: TCompareAccountsCard) => {
+    const { regulationFlags } = useRegulationFlags();
+    const { isEU } = regulationFlags;
+
+    return (
+        <div>
```
Comment (hasan-deriv) [inline src/cfd/screens/CFDCompareAccounts/CompareAccountsCard.tsx:26]: Is this `div` needed?

Comment (shontzu-deriv) [reply src/cfd/screens/CFDCompareAccounts/CompareAccountsCard.tsx:26]: yes

File: src/cfd/screens/CFDCompareAccounts/InstrumentsIconWithLabel.tsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+const InstrumentsIconWithLabel: FC<TInstrumentsIcon> = ({ highlighted, icon, isAsterisk, text }) => {
+    const InstrumentIcon = InstrumentsIcons[icon];
+    return (
+        <div
+            className='flex items-center m-2 cursor-not-allowed'
+            data-testid='dt_instruments_icon_container'
+            style={{
+                opacity: highlighted ? '' : '0.2',
+            }}
```
[the remark this answers is not in the document]
Comment (shontzu-deriv) [reply src/cfd/screens/CFDCompareAccounts/InstrumentsIconWithLabel.tsx:22]: yes

File: src/cfd/screens/CFDCompareAccounts/CompareAccountsCard.tsx (hunk 4 above)
Comment (shontzu-deriv) [reply src/cfd/screens/CFDCompareAccounts/CompareAccountsCard.tsx:26]: yes, it is a container
image

**Review by thisyahlen-deriv — APPROVED** [review]

**Review by hasan-deriv — APPROVED** [review]

**Pull request merged**